sdks/ruby/src/lib/hatchet/features/runs.rb

```diff
@@ -0,0 +1,423 @@
+# frozen_string_literal: true
+
+require 'time'
+require 'timeout'
+
+module Hatchet
+  module Features
+    # Filter options for listing task runs
+    class RunFilter
+      attr_accessor :since, :until_time, :statuses, :workflow_ids, :additional_metadata
+
+      def initialize(since:, until_time: nil, statuses: nil, workflow_ids: nil, additional_metadata: nil)
+        @since = since
+        @until_time = until_time
+        @statuses = statuses
+        @workflow_ids = workflow_ids
+        @additional_metadata = additional_metadata
+      end
+    end
+
+    # Options for bulk cancel and replay operations
+    class BulkCancelReplayOpts
+      attr_accessor :ids, :filters
+
+      def initialize(ids: nil, filters: nil)
+        raise ArgumentError, "ids or filters must be set" if !ids && !filters
+        raise ArgumentError, "ids and filters cannot both be set" if ids && filters
+        
+        @ids = ids
+        @filters = filters
+      end
+
+      def v1_task_filter
+        return nil unless @filters
+
+        HatchetSdkRest::V1TaskFilter.new(
+          since: @filters.since,
+          until: @filters.until_time,
+          statuses: @filters.statuses,
+          workflow_ids: @filters.workflow_ids,
+          additional_metadata: maybe_additional_metadata_to_kv(@filters.additional_metadata)
+        )
+      end
+
+      def to_cancel_request
+        HatchetSdkRest::V1CancelTaskRequest.new(
+          external_ids: @ids,
+          filter: v1_task_filter
+        )
+      end
+
+      def to_replay_request
+        HatchetSdkRest::V1ReplayTaskRequest.new(
+          external_ids: @ids,
+          filter: v1_task_filter
+        )
+      end
+
+      private
+
+      def maybe_additional_metadata_to_kv(metadata)
+        return nil unless metadata
+        metadata.map { |k, v| { key: k.to_s, value: v.to_s } }
```

Comment:
do we want to force these to strings?

Reply:
it won't work in the frontend if it's not a string though I don't think (unless pg has some magic type matching capabilities in json)